### PR TITLE
Configuration

### PR DESCRIPTION
**Pull request by NYLawal**
Added dependencies, configured knex with typescript, added database configuration and established database connection.

### CODE REVIEW

**Comment by horlaarsco** [comment]
Can you add your dest folder to git ignore. Compile JS file should not be pushed to github

File: src/database/knexfile.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    },
+};
+
+export default config;
+
+
+
+
+
+// npx knex migrate:make --migrations-directory ./src/database/migrations -x ts user
```
Comment (horlaarsco) [inline src/database/knexfile.ts]: Can you remove the comments?

File: src/database/migrations/20240806044802_user.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    return knex.schema
+    .createTable('users', function (table) {
+      table.increments('id');
+      table.string('first_name', 255).notNullable();
+      table.string('last_name', 255).notNullable();
+    })
+}
+
+
+export async function down(knex: Knex): Promise<void> {
```
Comment (horlaarsco) [inline src/database/migrations/20240806044802_user.ts:17]: You should put a down which would be the drop table command so when you rollback your migrations it removes the table

Comment (NYLawal) [reply src/database/migrations/20240806044802_user.ts:17]: command added

**Review by horlaarsco — APPROVED** [review]

**Pull request merged**